pandas/core/series.py

```diff
@@ -166,9 +166,9 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                     else:
                         index = Index(_try_sort(data))
                 try:
-                    if isinstance(index, DatetimeIndex):
+                    if isinstance(index, DatetimeIndex) and lib.infer_dtype(data) != 'datetime64':
```

Comment:
I ran the tests and reported the results in the issue: https://github.com/pydata/pandas/issues/9456.

Maybe a better fix would be to correctly handle the comparison between Timestamp and datetime64?